test/net-lookup.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const assert = require('bsert');
+const Network = require('../lib/protocol/network');
+const {lookup, resolve} = require('../lib/net/lookup');
+
+const main = Network.get('main');
+
+const notAHost = 'not-a-domain.not-a-domain';
+
+describe('Lookup', function() {
+  it('should lookup seed', async () => {
+    for (const host of main.seeds) {
+      const addresses = await lookup(host);
+      assert(addresses.length > 0, 'addresses not found.');
+    }
+  });
+
+  it('should fail lookup', async () => {
+    let err;
+
+    try {
+      await lookup(notAHost);
+    } catch (e) {
+      err = e;
+    }
+
+    assert(err);
+    assert.strictEqual(err.message, 'No DNS results.');
+  });
+
+  it('should lookup seed', async () => {
+    this.timeout(10000);
+
+    for (const host of main.seeds) {
+      const addresses = await resolve(host);
+
+      assert(addresses.length > 0, 'addresses not found.');
+    }
+  });
+
+  it('should fail resolve', async () => {
+    let err;
+
+    try {
+      await resolve(notAHost);
+    } catch (e) {
+      err = e;
+    }
+
+    assert(err);
+    assert.strictEqual(err.message, `Query error: NXDOMAIN (${notAHost} A).`);
+
+    // TODO: Host that does not have A/AAAA records?
+  });
```

Comment:
It is used in `hostlist.js` populate method to resolve DNS when connecting to DNS hosts.
It can be DNS Name of the `seed` or the `node` in the initial configuration.

Reply:
yeah but:

https://github.com/handshake-org/hsd/blob/107ed2be9280bb4292d21330e7d1ace62da9a002/lib/net/hostlist.js#L1626

I added this bug, can not get it to throw. (Although moving the bug to `lookup` does)

```diff
diff --git a/lib/net/lookup.js b/lib/net/lookup.js
index 4413671c1..91349265c 100644
--- a/lib/net/lookup.js
+++ b/lib/net/lookup.js
@@ -142,4 +142,4 @@ async function _resolve(name, family) {
  */

 exports.lookup = lookup;
-exports.resolve = resolve;
+exports.resolve = () => {throw new Error('Not Implemented.')};
```

Reply:
Okay, double checked. You are right recursive resolver is not used. Only stub resolver with set of servers is used.
I guess, it's more reliable to use those servers than use system resolver ?

So I don't see a reason to remove it right now, we could reuse it in the future.

p.s. You wont see errors even if you throw it from lookup, because those are enclosed with try/catch in populate. But, yeah they should see the error in the logs if you use `--nodes=example.org` for example.